[es] Add content/es/docs/concepts/storage/storage-classes.md (#41118)
kubernetes/website: content/es/docs/concepts/storage/storage-classes.md

---
reviewers:
  - edithturn
  - raelga
  - electrocucaracha

> title: Clases de Almacenamiento

Reviewer: Como en el resto del documento se ocupa _StorageClass_, podríamos poner la referencia igual en el título

```suggestion
title: StorageClass (Clases de Almacenamiento)
```

Author: Entiendo, tiene sentido :)
Gracias

content_type: concept
weight: 40
---

<!-- overview -->

> Este documento describe el concepto de StorageClass en Kubernetes. Necesita estar familiarizado con

Reviewer: Se podría mencionar el término en Inglés porque es más común solo para referencia entre paréntesis la primera vez.

```suggestion
Este documento describe el concepto de Clases de Almacenamiento (StorageClass) en Kubernetes. Necesita estar familiarizado con
```

> [volumes](/docs/concepts/storage/volumes/) y

Reviewer: ```suggestion
[volúmenes](/docs/concepts/storage/volumes/) y
```

> [persistent volumes](/docs/concepts/storage/persistent-volumes).

Reviewer: ```suggestion
[volúmenes persistentes](/docs/concepts/storage/persistent-volumes).
```

<!-- body -->

## Introducción

> StorageClass proporciona una forma para que los administradores describan las "clases" de

Reviewer: ```suggestion
Una Clase de Almacenamiento proporciona una forma para que los administradores describan las "clases" de
```

almacenamiento que ofrecen. Diferentes clases pueden corresponder a niveles de calidad de servicio,
o a políticas de copia de seguridad, o a políticas arbitrarias determinadas por los administradores del clúster
de Kubernetes. Kubernetes en sí no tiene opiniones sobre lo que representan las clases. Este concepto a veces se denomina "profiles" en otros sistemas de almacenamiento.

## El recurso StorageClass

Cada StorageClass contiene los campos `provisioner`, `parameters` y
`reclaimPolicy`, que se utilizan cuando un PersistentVolume que pertenece a
la clase debe aprovisionarse dinámicamente.

El nombre de un objeto StorageClass es significativo y es la forma en que los usuarios pueden
solicitar una clase en particular. Los administradores establecen el nombre y otros parámetros
de una clase al crear objetos StorageClass por primera vez, y los objetos no pueden
actualizarse una vez creados.

Los administradores pueden especificar un StorageClass predeterminada solo para los PVC que no

> solicite cualquier clase en particular a la que vincularse: vea el

Reviewer: ```suggestion
solicite cualquier clase en particular a la que vincularse: vea la
```

| Cinder                 |        &#x2713;        | [OpenStack Cinder](#openstack-cinder) |
| FC                     |           -            |                   -                   |
| FlexVolume             |           -            |                   -                   |
| GCEPersistentDisk      |        &#x2713;        |           [GCE PD](#gce-pd)           |
| iSCSI                  |           -            |                   -                   |
| NFS                    |           -            |              [NFS](#nfs)              |
| RBD                    |        &#x2713;        |         [Ceph RBD](#ceph-rbd)         |
| VsphereVolume          |        &#x2713;        |          [vSphere](#vsphere)          |
| PortworxVolume         |        &#x2713;        |  [Portworx Volume](#portworx-volume)  |
| Local                  |           -            |            [Local](#local)            |

No está restringido especificar los aprovisionadores "internos"
enumerados aquí (cuyos nombres tienen el prefijo "kubernetes.io" y se envían
junto con Kubernetes). También puede ejecutar y especificar aprovisionadores externos,

> que son programas independientes que siguen un [especificación](https://git.k8s.io/design-proposals-archive/storage/volume-provisioning.md)

Reviewer: ```suggestion
que son programas independientes que siguen una [especificación](https://git.k8s.io/design-proposals-archive/storage/volume-provisioning.md)
```

> definido por Kubernetes. Los autores de proveedores externos tienen total discreción

Reviewer: ```suggestion
definida por Kubernetes. Los autores de proveedores externos tienen total discreción
```

> sobre dónde vive su código, cómo se envía el aprovisionador, cómo debe ser
> ejecutada, qué complemento de volumen usa (incluido Flex), etc. El repositorio

Reviewer 2: Me [parece](https://www.diccionariodedudas.com/donde-o-donde/) que no llevan acento
```suggestion
sobre donde vive su código, como se envía el aprovisionador, como debe ser
ejecutada, que complemento de volumen usa (incluido Flex), etc. El repositorio
```

Author: Estuve con dudas, y encontré algo más claro aqui:
`Se escribe donde, sin tilde, cuando podemos equiparar la función que tiene la oración que encabeza con un complemento de lugar («Yoli trabaja cerca de donde vive Paco»); pero se escribe con tilde en las oraciones interrogativas en las que se pregunta por el lugar («¿Dónde trabaja Yoli?» o «No estoy segura de dónde trabaja Yoli») y en las exclamativas («¡Dónde hemos ido a parar!»).`

https://www.fundeu.es/recomendacion/dondedonde/#:~:text=Se%20escribe%20donde%2C%20sin%20tilde,No%20estoy%20segura%20de%20d%C3%B3nde
@electrocucaracha

Reviewer 2: En ocaciones es algo confuso. Me parece que tienes razon, quiza en este caso sea necesario ser mas enfaticos.

En su lugar, puede usar el selector de nodos para el nombre de host en este caso, como se muestra a continuación.
{{< /note >}}

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: task-pv-pod
spec:
  nodeSelector:
    kubernetes.io/hostname: kube-01
  volumes:
    - name: task-pv-storage
      persistentVolumeClaim:
        claimName: task-pv-claim
  containers:
    - name: task-pv-container
      image: nginx
      ports:
        - containerPort: 80
          name: "http-server"
      volumeMounts:
        - mountPath: "/usr/share/nginx/html"
          name: task-pv-storage
```

### Topologías permitidas

Cuando un operador de clúster especifica el modo de enlace de volumen `WaitForFirstConsumer`, ya no es necesario restringir el aprovisionamiento a topologías específicas en la mayoría de las situaciones. Sin embargo, si todavía es necesario, se puede especificar `allowedTopologies`.

This example demonstrates how to restrict the topology of provisioned volumes to specific
zones and should be used as a replacement for the `zone` and `zones` parameters for the

> supported plugins.

Reviewer 2: Por traducir?

Author: Ups, lo olvidé :)
Ya lo envié, @electrocucaracha 
Thank youu!

provisioner: kubernetes.io/gce-pd
parameters:
  type: pd-standard
volumeBindingMode: WaitForFirstConsumer
allowedTopologies:
  - matchLabelExpressions:
      - key: failure-domain.beta.kubernetes.io/zone
        values:
          - us-central-1a
          - us-central-1b
```

## Parámetros

Las clases de almacenamiento tienen parámetros que describen los volúmenes que pertenecen a la clase de almacenamiento. Se pueden aceptar diferentes parámetros dependiendo del `provisioner`. Por ejemplo, el valor `io1`, para el parámetro `type` y el parámetro `iopsPerGB` son específicos de EBS. Cuando se omite un parámetro, se utiliza algún valor predeterminado.

Puede haber como máximo 512 parámetros definidos para StorageClass.
La longitud total del objeto de parámetros, incluidas sus claves y valores, no puede superar los 256 KiB.

### AWS EBS

```yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: slow
provisioner: kubernetes.io/aws-ebs
parameters:
  type: io1
  iopsPerGB: "10"
  fsType: ext4
```

- `type`: `io1`, `gp2`, `sc1`, `st1`. Ver
  [AWS docs](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSVolumeTypes.html)
  para detalles. Por defecto: `gp2`.

> - `zone` (Obsoleto): AWS sona. Si no se especifica `zone` ni `zones`, los volúmenes generalmente se distribuyen por turnos en todas las zonas activas donde el clúster de Kubernetes tiene un nodo. Los parámetros `zone` y `zones` no deben usarse al mismo tiempo.

Reviewer 2: ```suggestion
- `zone` (Obsoleto): AWS zona. Si no se especifica `zone` ni `zones`, los volúmenes generalmente se distribuyen por turnos en todas las zonas activas donde el clúster de Kubernetes tiene un nodo. Los parámetros `zone` y `zones` no deben usarse al mismo tiempo.
```

Aquí hay unos ejemplos:

- [Servidor NFS Ganesha y aprovisionador externo](https://github.com/kubernetes-sigs/nfs-ganesha-server-and-external-provisioner)
- [Aprovisionador externo de subdirección NFS](https://github.com/kubernetes-sigs/nfs-subdir-external-provisioner)

### OpenStack Cinder

```yaml
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: gold
provisioner: kubernetes.io/cinder
parameters:
  availability: nova
```

- `availability`: Zona de disponibilidad. Si no se especifica, los volúmenes generalmente se distribuyen por turnos en todas las zonas activas donde el clúster de Kubernetes tiene un nodo.

{{< note >}}
{{< feature-state state="deprecated" for_k8s_version="v1.11" >}}
Este proveedor interno de OpenStack está obsoleto. Por favor use [el proveedor de nube externo para OpenStack](https://github.com/kubernetes/cloud-provider-openstack).
{{< /note >}}

### vSphere

Hay dos tipos de aprovisionadores para las clases de almacenamiento de vSphere:

- [CSI provisioner](#vsphere-provisioner-csi): `csi.vsphere.vmware.com`
- [vCP provisioner](#vcp-provisioner): `kubernetes.io/vsphere-volume`

> Los proveedores In-tree son [deprecated](/blog/2019/12/09/kubernetes-1-17-feature-csi-migration-beta/#why-are-we-migrating-in-tree-plugins-to-csi). Para obtener más información sobre el aprovisionador de CSI, consulte [Kubernetes vSphere CSI Driver](https://vsphere-csi-driver.sigs.k8s.io/) y [vSphereVolume CSI migration](/docs/concepts/storage/volumes/#vsphere-csi-migration).

Reviewer 2: ```suggestion
Los proveedores In-tree estan [obsoletos](/blog/2019/12/09/kubernetes-1-17-feature-csi-migration-beta/#why-are-we-migrating-in-tree-plugins-to-csi). Para obtener más información sobre el aprovisionador de CSI, consulte [Kubernetes vSphere CSI Driver](https://vsphere-csi-driver.sigs.k8s.io/) y [vSphereVolume CSI migration](/docs/concepts/storage/volumes/#vsphere-csi-migration).
```

Los siguientes ejemplos utilizan el aprovisionador StorageClass de VMware Cloud Provider (vCP).

1. Cree una StorageClass con un formato de disco especificado por el usuario.

   ```yaml
   apiVersion: storage.k8s.io/v1
   kind: StorageClass
   metadata:
     name: fast
   provisioner: kubernetes.io/vsphere-volume
   parameters:
     diskformat: zeroedthick
   ```

   `diskformat`: `thin`, `zeroedthick` y `eagerzeroedthick`. Por defecto: `"thin"`.

2. Cree una StorageClass con un formato de disco en un almacén de datos especificado por el usuario.

   ```yaml
   apiVersion: storage.k8s.io/v1
   kind: StorageClass
   metadata:
     name: fast
   provisioner: kubernetes.io/vsphere-volume
   parameters:
     diskformat: zeroedthick
     datastore: VSANDatastore
   ```

   `datastore`: el usuario también puede especificar el almacén de datos en StorageClass. El volumen se creará en el almacén de datos especificado en StorageClass, que en este caso es `VSANDatastore`. Este campo es opcional. Si no se especifica el almacén de datos, el volumen se creará en el almacén de datos especificado en el archivo de configuración de vSphere utilizado para inicializar vSphere Cloud Provider.

> 3. Gestión de políticas de almacenamiento dentro de kubernetes

Reviewer 2: nit
```suggestion
3. Gestión de políticas de almacenamiento dentro de Kubernetes
```

     Las políticas de SPBM se pueden especificar en StorageClass mediante el parámetro `storagePolicyName`.

   - Soporte de políticas Virtual SAN dentro de Kubernetes

     Los administradores de Vsphere Infrastructure (VI) tendrán la capacidad de especificar capacidades de almacenamiento Virtual SAN personalizadas durante el aprovisionamiento dinámico de volúmenes. Ahora puede definir los requisitos de almacenamiento, como el rendimiento y la disponibilidad, en forma de capacidades de almacenamiento durante el aprovisionamiento dinámico de volúmenes.
     Los requisitos de capacidad de almacenamiento se convierten en una política de Virtual SAN que luego se transfiere a la capa de Virtual SAN cuando se crea un volumen persistente (disco virtual). El disco virtual se distribuye en el almacén de datos de Virtual SAN para cumplir con los requisitos.

>      Puedes ver[Administración basada en políticas de almacenamiento para el aprovisionamiento dinámico de volúmenes](https://github.com/vmware-archive/vsphere-storage-for-kubernetes/blob/fa4c8b8ad46a85b6555d715dd9d27ff69839df53/documentation/policy-based-mgmt.md)

Reviewer 2: ```suggestion
     Puedes ver la [Administración basada en políticas de almacenamiento para el aprovisionamiento dinámico de volúmenes](https://github.com/vmware-archive/vsphere-storage-for-kubernetes/blob/fa4c8b8ad46a85b6555d715dd9d27ff69839df53/documentation/policy-based-mgmt.md)
```

- `adminId`: ID de cliente de Ceph que es capaz de crear imágenes en el grupo.
  El valor predeterminado es "admin".
- `adminSecretName`: Nombre secreto para `adminId`. Este parámetro es obligatorio. El secreto proporcionado debe tener el tipo "kubernetes.io/rbd".
- `adminSecretNamespace`: El espacio de nombres para `adminSecretName`. El valor predeterminado es "predeterminado".
- `pool`: Grupo Ceph RBD. El valor predeterminado es "rbd".
- `userId`: ID de cliente de Ceph que se utiliza para asignar la imagen RBD. El valor predeterminado es el mismo que `adminId`.
- `userSecretName`: El nombre de Ceph Secret para `userId` para mapear la imagen RBD. Él
  debe existir en el mismo espacio de nombres que los PVC. Este parámetro es obligatorio.
  El secreto proporcionado debe tener el tipo "kubernetes.io/rbd", por ejemplo creado de esta manera:

  ```shell
  kubectl create secret generic ceph-secret --type="kubernetes.io/rbd" \
    --from-literal=key='QVFEQ1pMdFhPUnQrSmhBQUFYaERWNHJsZ3BsMmNjcDR6RFZST0E9PQ==' \
    --namespace=kube-system
  ```

- `userSecretNamespace`: El espacio de nombres para `userSecretName`.

> - `fsType`: fsType que es compatible con kubernetes. Por defecto: `"ext4"`.

Reviewer 2: ```suggestion
- `fsType`: fsType que es compatible con Kubernetes. Por defecto: `"ext4"`.
```

parameters:
  skuName: Standard_LRS
  location: eastus
  storageAccount: azure_storage_account_name
```

- `skuName`: Nivel de SKU de la cuenta de almacenamiento de Azure. El valor predeterminado está vacío.
- `location`: Ubicación de la cuenta de almacenamiento de Azure. El valor predeterminado está vacío.
- `storageAccount`: Nombre de la cuenta de almacenamiento de Azure. El valor predeterminado está vacío. Si un almacenamiento
  no se proporciona la cuenta, se buscan todas las cuentas de almacenamiento asociadas con el grupo de recursos para encontrar una que coincida con `skuName` y `location`. Si se proporciona una cuenta de almacenamiento, debe residir en el mismo grupo de recursos que el clúster y se ignoran `skuName` y `location`.
- `secretNamespace`: el espacio de nombres del secreto que contiene el nombre y la clave de la cuenta de Azure Storage. El valor predeterminado es el mismo que el Pod.
- `secretName`: el nombre del secreto que contiene el nombre y la clave de la cuenta de Azure Storage. El valor predeterminado es `azure-storage-account-<accountName>-secret`
- `readOnly`: a flag indicating whether the storage will be mounted as read only.

>   Defaults to false which means a read/write mount. This setting will impact theuna bandera que indica si el almacenamiento se montará como de solo lectura. El valor predeterminado es falso, lo que significa un montaje de lectura/escritura. Esta configuración también afectará la configuración `ReadOnly` en VolumeMounts.

Reviewer 2: Por traducir

Author: Thank you, @electrocucaracha :)
